codegen/src/test/resources/software/amazon/awssdk/codegen/poet/paginators/PaginatedOperationWithoutResultKeyPublisher.java

```diff
@@ -0,0 +1,130 @@
+ *
+ * public void onSubscribe(org.reactivestreams.Subscriber subscription) { //... };
+ *
+ *
+ * public void onNext(software.amazon.awssdk.services.jsonprotocoltests.model.PaginatedOperationWithoutResultKeyResponse response) { //... };
+ * });}
+ * </pre>
+ *
+ * As the response is a publisher, it can work well with third party reactive streams implementations like RxJava2.
+ * <p>
+ * <b>Note: If you prefer to have control on service calls, use the
+ * {@link #paginatedOperationWithoutResultKey(software.amazon.awssdk.services.jsonprotocoltests.model.PaginatedOperationWithoutResultKeyRequest)}
+ * operation.</b>
+ * </p>
+ */
+@Generated("software.amazon.awssdk:codegen")
+public class PaginatedOperationWithoutResultKeyPublisher implements SdkPublisher<PaginatedOperationWithoutResultKeyResponse> {
+    private final JsonProtocolTestsAsyncClient client;
+
+    private final PaginatedOperationWithoutResultKeyRequest firstRequest;
+
+    private final AsyncPageFetcher nextPageFetcher;
+
+    private boolean isLastPage;
+
+    public PaginatedOperationWithoutResultKeyPublisher(final JsonProtocolTestsAsyncClient client,
+                                                       final PaginatedOperationWithoutResultKeyRequest firstRequest) {
+        this.client = client;
+        this.firstRequest = firstRequest;
+        this.nextPageFetcher = new PaginatedOperationWithoutResultKeyResponseFetcher();
+    }
+
+    @Override
+    public void subscribe(Subscriber<? super PaginatedOperationWithoutResultKeyResponse> subscriber) {
+        subscriber.onSubscribe(new ResponsesSubscription(subscriber, nextPageFetcher));
+    }
+
+    /**
+     * <p>
+     * A helper method to resume the pages in case of unexpected failures. The method takes the last successful response
+     * page as input and returns an instance of {@link PaginatedOperationWithoutResultKeyPublisher} that can be used to
+     * retrieve the consecutive pages that follows the input page.
+     * </p>
+     */
+    public PaginatedOperationWithoutResultKeyPublisher resume(final PaginatedOperationWithoutResultKeyResponse lastSuccessfulPage) {
+        if (nextPageFetcher.hasNextPage(lastSuccessfulPage)) {
+            return new PaginatedOperationWithoutResultKeyPublisher(client, firstRequest.toBuilder()
+                                                                                       .nextToken(lastSuccessfulPage.nextToken()).build());
+        }
+        return new PaginatedOperationWithoutResultKeyPublisher(client, firstRequest) {
+            @Override
+            public void subscribe(Subscriber<? super PaginatedOperationWithoutResultKeyResponse> subscriber) {
+                subscriber.onSubscribe(new EmptySubscription(subscriber));
+            }
+        }.withLastPage(true);
+    }
+
+    PaginatedOperationWithoutResultKeyPublisher withLastPage(boolean isLastPage) {
```

Comment:
Why do you need a setter for this? Can't you just set it in the constructor?

Reply:
It can be set in constructor too. But having it in setter might hide it better constructor.

Reply:
Actually thinking again, having a private constructor is better than the setter. Will fix it.